internal/client/amconn.go

```diff
@@ -242,6 +247,80 @@ func (c *amConnection) getServerInfo() (info serverInfo, err error) {
 	return info, err
 }
 
+// getJWKSURI gets the OAuth 2.0 JSON Web Key set URI from AM
+func (c *amConnection) getJWKSURI() (uri string, err error) {
+	request, err := http.NewRequest(
+		http.MethodGet,
+		c.baseURL+"/oauth2/.well-known/openid-configuration?realm="+c.realm,
+		nil)
+	if err != nil {
+		debug.Logger.Println(debug.DumpHTTPRoundTrip(request, nil))
+		return uri, err
+	}
+
+	request.Header.Add(httpContentType, string(ApplicationJSON))
+	response, err := c.Do(request)
+	if err != nil {
+		debug.Logger.Println(debug.DumpHTTPRoundTrip(request, response))
+		return uri, err
+	}
+	defer response.Body.Close()
+	responseBody, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		debug.Logger.Println(debug.DumpHTTPRoundTrip(request, response))
+		return uri, err
+	}
+	if response.StatusCode != http.StatusOK {
+		debug.Logger.Println(debug.DumpHTTPRoundTrip(request, response))
+		return uri, fmt.Errorf("openid-configuration request failed")
+	}
+	var config struct {
+		URI string `json:"jwks_uri"`
+	}
+	if err = json.Unmarshal(responseBody, &config); err != nil {
+		debug.Logger.Println(debug.DumpHTTPRoundTrip(request, response))
+		return uri, err
+	}
+	return config.URI, err
+}
+
+// getJWKS gets the OAuth 2.0 JSON Web Key set from AM
+func (c *amConnection) getJWKS() (err error) {
+	if c.jwksURI == "" {
```

Comment:
I don't think we should cache the JWKS URI, because it would require a client/gateway restart if the AM config is changed.